packages/swingset-theme-hashicorp/src/components/side-nav/category.tsx

```diff
@@ -0,0 +1,99 @@
+import { NavigationNode, ComponentNode } from 'swingset/types'
+import { LinkItem } from './link-item'
+import { cx } from 'class-variance-authority'
+
+function Category({
+  title,
+  items,
+}: {
+  title: string
+  items: NavigationNode[]
+}) {
+  return (
+    <li className="ss-list-none">
+      <section className="ss-mb-4">
+        <CategoryHeading>{title.toUpperCase()}</CategoryHeading>
+        <ComponentList items={items} />
+      </section>
+    </li>
+  )
+}
+//Swap this out for already existing heading && Enquire about semantics, https://helios.hashicorp.design/components/application-state uses <div>
+function CategoryHeading({ children }: { children: string }) {
+  return (
+    <div className="ss-uppercase ss-text-xs ss-font-semibold ss-leading-6 ss-text-foreground-faint ss-border-b ss-border-faint ss-pb-2">
+      {children.toUpperCase()}
+    </div>
+  )
+}
+
+function ComponentList({
+  isNested,
+  items,
+}: {
+  isNested?: true
+  items: NavigationNode[]
+}) {
+  return (
+    <ul
+      className={cx(
+        'ss-mt-2 ss-space-y-1',
+        isNested && 'ss-ml-2 ss-border-l ss-border-faint ss-mt-0'
+      )}
+    >
+      {items.map((item) => {
+        const isFolder = item.__type === 'folder'
```

Comment:
This could be a good candidate for using a switch statement on `__type`. It's stylistic really, but I think it would make the logic of this block easier to understand.